indico/modules/events/registration/tasks.py

```diff
@@ -58,3 +58,19 @@ def delete_field_data():
                     logger.error('Failed to delete file for %r: %s', data, e)
             data.field_data.field.is_purged = True
     db.session.commit()
+
+
+@celery.periodic_task(name='registration_retention_period', run_every=crontab(minute='0', hour='3'))
+def delete_registration():
+    registrations = (Registration.query
+                     .join(RegistrationForm)
+                     .join(Event)
+                     .filter(RegistrationForm.retention_period.isnot(None),
+                             Event.end_dt + RegistrationForm.retention_period <= now_utc())
+                     .all())
+
+    for reg in registrations:
+        logger.info('Purging registration: %r', reg)
+        reg.delete()
```

Comment:
- delete files explicitly (just go over all field data of the registration that has a `storage_file_id`, no need to check field types etc)
- `db.session.delete(...)` to delete the registration itself